### DISCUSSION TITLE
Merge authors or releases?

### DISCUSSION THREAD

**Discussion by bbbhltz** [bookwyrm-social/bookwyrm#2076] [Q&A]
I was wondering if perhaps I am missing something. Is there a way to merge authors or releases? I seem to come across duplicates from time to time and I am uncertain if there is a way to take care of this?

I will note that in several cases, there are also duplicates on OpenLibrary.

**Comment by mouse-reeve** [comment]
Short answer, not at the moment. Sorry about that!

Long answer: There isn't a way to merge them from the UI, although I'd like there to be! It isn't something I've gotten to yet. I do have a management command that can automatically merge duplicate entries, which works if the items have shared identifiers (like ISBN or ISNI), but at the moment it has to be run manually. And you're right that the source of the problem is often OpenLibrary duplicates -- another thing I've been meaning to do is allow entries to have multiple OpenLibrary IDs so that they can reflect duplicates there.

**Comment by thalskarth** [comment]
A way to merge duplicated books (as different editions) would me great. It's very common to find books in different languages as diferente books instead of editions.

I doesn't have to be automated, a manual way to just check and merge to books into 1 it's more than usefull.

**Comment by placoderm** [comment]
I just added a bunch of books with the same author but it ended up creating a new author for each one (despite choosing the existing author). I'd love to be able to merge them.

**Comment by jaschaurbach** [comment]
When I had the problem with the a new author for every book I took the first one created on the instance (the one with the smallest ID) and changed every book to this author. To accomplish this, you need to edit the book, tick "delete author" and add the name, hit save. In the next window choose the one you want to use (you can check the ID by hovering over to see the link in your browser status bar). repeat till all books are attached to this author.

**Comment by LokiMidgard** [comment]
This would be a nice feature. I accidently added a new Author when I imported a book from [Inventaire](https://inventaire.io/).

**Comment by GarrickWinter** [comment]
I've been wrangling importing my book shelves from Goodreads, and in the process of entering manual data for books that weren't matched, I'm finding that a significant proportion of books seem to have duplicates that should be merged; often there are reviews on at least two of the versions. Sometimes the books are orphaned or assigned to duplicate authors, too, so they appear in title searches but not via author searches. I'd be happy to volunteer some time to flag these and help with the cleanup if there was some mechanism for doing so.

**Comment by codeinabox** [comment]
I would love a feature to merge books or authors. As much as I love the fact that BookWyrm is powered by open data, it also means it inherits the duplicate issues on Open Library. It is possible to [apply to be an Open Library librarian](https://openlibrary.org/librarians-in-training), so you have elevated privileges to merge duplicates - perhaps BookWyrm could have something similar?

**Reply by mouse-reeve** [reply]
This is in progress in https://github.com/bookwyrm-social/bookwyrm/pull/3914
